File: LeetCode/LongestPrefix/LongestCommonPrefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {

       var len = strs.Length;
       if(len<2){
        return strs[0];
       }
       var str1 = strs[0];
       var str2 = strs[1];
       var lngPrefix = "";
       var shortString = str1.Length<str2.Length ? str1 : str2;
       
       for(int i = 0; i<shortString.Length; i++){
        if(str1[i] == str2[i]){
            lngPrefix += str1[i];
        }
        else{
            break;
        }
       }

       if(lngPrefix!="" && len>2){
        for(int i = 2; i<len; i++){
                var strToCheck = strs[i];
                int limit = Math.Min(lngPrefix.Length, strToCheck.Length);

                int j = 0;
                while (j < limit && strToCheck[j] == lngPrefix[j]) {
                j++;
                }

                lngPrefix = lngPrefix.Substring(0, j);
                if (lngPrefix == "") break;

            }
        }

       return lngPrefix;
    }
}

File: LeetCode/LongestPrefix/LongestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
  let len = strs.length;
  if (len < 2) {
    return strs[0];
  }
  let str1 = strs[0];
  let str2 = strs[1];

  let lngPrefixStr = "";
  let shorterStrLen = str1.length < str2.length ? str1.length : str2.length;

  for (let i = 0; i < shorterStrLen; i++) {
    if (str1[i] === str2[i]) {
      lngPrefixStr += str1[i];
    } else {
      break;
    }
  }
  if (lngPrefixStr != "" && len > 2) {
    for (let i = 2; i < strs.length; i++) {
      let strToCheck = strs[i];
      let limit = Math.min(lngPrefixStr.length, strToCheck.length);
      let j = 0;
      while (j < limit && lngPrefixStr[j] == strToCheck[j]) {
        j++;
      }
      lngPrefixStr = lngPrefixStr.slice(0, j);
      if (lngPrefixStr == "") break;
    }
  }

  return lngPrefixStr;
}
